src/glyph.rs

```diff
@@ -0,0 +1,166 @@
+        unsafe { from_glib_none((*self.to_glib_none().0).glyphs) }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct GlyphInfo(*mut pango_sys::PangoGlyphInfo);
+
+impl GlyphInfo {
+    pub fn glyph(&self) -> u32 {
+        unsafe { (*self.0).glyph }
+    }
+
+    pub fn geometry(&self) -> &GlyphGeometry {
+        unsafe { &*(&((*self.0).geometry) as *const _ as *const GlyphGeometry) }
+    }
+}
+
+impl FromGlibContainerAsVec<*mut pango_sys::PangoGlyphInfo, *mut pango_sys::PangoGlyphInfo>
+    for GlyphInfo
+{
+    unsafe fn from_glib_none_num_as_vec(
+        ptr: *mut pango_sys::PangoGlyphInfo,
+        num: usize,
+    ) -> Vec<Self> {
+        if num == 0 || ptr.is_null() {
+            return Vec::new();
+        }
+        let mut res = Vec::with_capacity(num);
+        for x in 0..num {
+            res.push(from_glib_none(ptr.add(x)));
+        }
+        res
+    }
+
+    unsafe fn from_glib_container_num_as_vec(
+        ptr: *mut pango_sys::PangoGlyphInfo,
+        num: usize,
+    ) -> Vec<Self> {
+        let res = FromGlibContainerAsVec::from_glib_none_num_as_vec(ptr, num);
+        glib_sys::g_free(ptr as *mut _);
+        res
+    }
+
+    unsafe fn from_glib_full_num_as_vec(
+        ptr: *mut pango_sys::PangoGlyphInfo,
+        num: usize,
+    ) -> Vec<Self> {
+        FromGlibContainerAsVec::from_glib_container_num_as_vec(ptr, num)
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtr<'a, *mut pango_sys::PangoGlyphInfo> for &'a GlyphInfo {
+    type Storage = &'a GlyphInfo;
+
+    fn to_glib_none(&self) -> Stash<'a, *mut pango_sys::PangoGlyphInfo, Self> {
+        Stash(self.0, *self)
+    }
+}
+
+#[doc(hidden)]
+impl<'a> ToGlibPtrMut<'a, *mut pango_sys::PangoGlyphInfo> for GlyphInfo {
+    type Storage = &'a mut Self;
+
+    #[inline]
+    fn to_glib_none_mut(&'a mut self) -> StashMut<'a, *mut pango_sys::PangoGlyphInfo, Self> {
+        StashMut(self.0, self)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrNone<*mut pango_sys::PangoGlyphInfo> for GlyphInfo {
+    unsafe fn from_glib_none(ptr: *mut pango_sys::PangoGlyphInfo) -> Self {
+        assert!(!ptr.is_null());
+        GlyphInfo(ptr)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrFull<*mut pango_sys::PangoGlyphInfo> for GlyphInfo {
+    unsafe fn from_glib_full(ptr: *mut pango_sys::PangoGlyphInfo) -> Self {
+        assert!(!ptr.is_null());
+        GlyphInfo(ptr)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrNone<*const pango_sys::PangoGlyphInfo> for GlyphInfo {
+    unsafe fn from_glib_none(ptr: *const pango_sys::PangoGlyphInfo) -> Self {
+        assert!(!ptr.is_null());
+        GlyphInfo(ptr as *mut _)
+    }
+}
+
+#[doc(hidden)]
+impl FromGlibPtrFull<*const pango_sys::PangoGlyphInfo> for GlyphInfo {
+    unsafe fn from_glib_full(ptr: *const pango_sys::PangoGlyphInfo) -> Self {
+        assert!(!ptr.is_null());
+        GlyphInfo(ptr as *mut _)
+    }
+}
+
+#[repr(C)]
+pub struct GlyphGeometry(pango_sys::PangoGlyphGeometry);
```

Comment:
Maybe (and elsewhere) derive at least Debug here, or impl Debug in a more descriptive way

Reply:
Added a derive, thanks